src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -150,6 +150,9 @@ impl LogicalAgg {
         } else {
             let exchange = RequiredDist::shard_by_key(input_col_num, self.group_key())
                 .enforce_if_not_satisfies(local_agg.into(), &Order::any())?;
+            // Local phase should have reordered the group keys into their required order.
+            // we can just follow it.
+            let group_key = (0..self.group_key().len()).collect();
```

Comment:
Is this assumption correct? IIRC there was something like `o2i_col_mapping`. Is that relevant here? Or any other interface methods I should use for this?

Reply:
I think your code is correct and the previous one is wrong.